pkg/segment/writer/metrics/metricssegment.go

```diff
@@ -1698,3 +1753,64 @@ func FindTagValuesUnrotated(tRange *dtu.MetricsTimeRange, mQuery *structs.Metric
 
 	return nil
 }
+
+func (mb *MetricsBlock) appendToWALBuffer(dp wal.WalDatapoint) error {
```

Comment:
continuing the comment from line 557, then in this function, you will get those 3 params. 

you would then just do 
```
mb.walDPBuffer[dpIdx].Timestamp = ts
mb.walDPBuffer[dpIdx].dpVal = dpVal
mb.walDPBuffer[dpIdx].tsid = tsid
dpIdx++
```
THis will improve mem efficiency, and keep reusing the buffer